### PR TITLE
Add CREATE RDF GRAPH ddl statement

### PR DESCRIPTION
**Pull request by andyfengHKU**
This is our first PR migrating RDF querying from the prototype implemented by @gaurav8297. It implements

- CREATE RDF GRAPH statement
- refactor create table clause, bound clause & logical operator. We no longer differentiate node/rel/rdf in these classes.

### CODE REVIEW

File: src/catalog/property.cpp [hunk 1]
```diff
@@ -53,5 +53,14 @@ std::vector<std::unique_ptr<catalog::Property>> Property::copyProperties(
     return propertiesToReturn;
 }
 
+std::vector<std::unique_ptr<Property>> Property::copy(
```
Comment (acquamarin) [inline src/catalog/property.cpp:56]: I think we have 
`std::vector<std::unique_ptr<catalog::Property>> Property::copyProperties(
    const std::vector<std::unique_ptr<catalog::Property>>& propertiesToCopy)`
Which does the same thing

File: src/binder/bind/bind_ddl.cpp [hunk 2]
```diff
@@ -24,40 +22,94 @@ namespace binder {
...
+    auto relMultiplicity = getRelMultiplicityFromString(extraInfo->relMultiplicity);
+    auto boundExtraInfo = std::make_unique<BoundRelExtraCreateTableInfo>(
+        relMultiplicity, srcTableID, dstTableID, srcPkDataType->copy(), dstPkDataType->copy());
+    auto boundCreateInfo = std::make_unique<BoundCreateTableInfo>(
+        tableName, std::move(boundProperties), std::move(boundExtraInfo));
+    return make_unique<BoundCreateTable>(
+        StatementType::CREATE_REL_TABLE, tableName, std::move(boundCreateInfo));
+}
+
+static constexpr char RDF_IRI[] = "_IRI";
```
Comment (acquamarin) [inline src/binder/bind/bind_ddl.cpp:79]: Move them to InternalKeyWord?

File: src/binder/bind/bind_ddl.cpp [hunk 3]
```diff
@@ -24,40 +22,94 @@ namespace binder {
...
+        tableName, std::move(boundProperties), std::move(boundExtraInfo));
+    return make_unique<BoundCreateTable>(
+        StatementType::CREATE_NODE_TABLE, tableName, std::move(boundCreateInfo));
 }
 
 std::unique_ptr<BoundStatement> Binder::bindCreateRelTableClause(
     const parser::Statement& statement) {
-    auto& createRelClause = (CreateRelClause&)statement;
-    auto tableName = createRelClause.getTableName();
+    auto& createTableClause = (CreateTableClause&)statement;
```
Comment (acquamarin) [inline src/binder/bind/bind_ddl.cpp:49]: Better to use reinterpret_cast

File: src/binder/bind/bind_ddl.cpp [hunk 4]
```diff
@@ -24,40 +22,94 @@ namespace binder {
...
+static constexpr char RDF_NODE_TABLE_SUFFIX[] = "_RESOURCE";
+static constexpr char RDF_REL_TABLE_SUFFIX[] = "_TRIPLES";
+
+std::unique_ptr<BoundStatement> Binder::bindCreateRdfGraphClause(
+    const parser::Statement& statement) {
+    auto& createTableClause = (CreateTableClause&)statement;
+    auto rdfGraphName = createTableClause.getTableName();
+    auto stringType = std::make_unique<LogicalType>(LogicalTypeID::STRING);
+    // RDF node (resource) table
+    auto nodeTableName = rdfGraphName + RDF_NODE_TABLE_SUFFIX;
```
Comment (acquamarin) [inline src/binder/bind/bind_ddl.cpp:90]: I think we can refactor these to a function:
1. getRdfGraphNodeTableName()
2. getRdfGraphRelTableName()
Which internally appends the suffix to the rdfGraphName

File: test/test_files/rdf/ddl.test [hunk 5]
```diff
@@ -0,0 +1,14 @@
+-GROUP Rdf
+-DATASET CSV tinysnb
+
+--
+
+-CASE CreateRdfGraph
```
Comment (acquamarin) [inline test/test_files/rdf/ddl.test:6]: Can you add more tests?
1. Execute create rdf inside an active manual transaction.
2. Test whether the table exists after the transaction is aborted.

File: src/storage/store/table_statistics.cpp [hunk 6]
```diff
@@ -13,7 +13,11 @@ TablesStatistics::TablesStatistics() {
 }
 
 void TablesStatistics::readFromFile(const std::string& directory) {
-    auto filePath = getTableStatisticsFilePath(directory, DBFileType::ORIGINAL);
```
Comment (acquamarin) [inline src/storage/store/table_statistics.cpp:16]: Maybe u can remove this api, and give dbFileType a default parameter
void TablesStatistics::readFromFile(const std::string& directory, common::DBFileType dbFileType = DBFileType::ORIGINAL)

File: src/storage/wal_replayer.cpp [hunk 7]
```diff
@@ -180,13 +183,13 @@ void WALReplayer::replayNodeTableRecord(const kuzu::storage::WALRecord& walRecor
...
-void WALReplayer::replayRelTableRecord(const kuzu::storage::WALRecord& walRecord) {
+void WALReplayer::replayRelTableRecord(const kuzu::storage::WALRecord& walRecord, bool isRdf) {
     if (isCheckpoint) {
         // See comments for NODE_TABLE_RECORD.
-        auto nodesStatisticsAndDeletedIDsForCheckPointing =
-            std::make_unique<NodesStatisticsAndDeletedIDs>(wal->getDirectory());
-        auto maxNodeOffsetPerTable =
-            nodesStatisticsAndDeletedIDsForCheckPointing->getMaxNodeOffsetPerTable();
+        auto fileType = isRdf ? DBFileType::WAL_VERSION : DBFileType::ORIGINAL;
+        auto nodesStatistics =
```
Comment (acquamarin) [inline src/storage/wal_replayer.cpp:190]: Inline the fileType

File: src/include/catalog/table_schema.h [hunk 8]
```diff
@@ -239,5 +240,35 @@ class RelTableSchema : public TableSchema {
...
+          nodeTableID{nodeTableID}, relTableID{relTableID} {}
+    RdfTableSchema(common::table_id_t nodeTableID, common::table_id_t relTableID)
+        : TableSchema{common::InternalKeyword::ANONYMOUS, common::INVALID_TABLE_ID, TableType::RDF,
+              std::vector<std::unique_ptr<Property>>{}},
+          nodeTableID{nodeTableID}, relTableID{relTableID} {}
+
+    inline common::table_id_t getNodeTableID() const { return nodeTableID; }
+    inline common::table_id_t getRelTableID() const { return relTableID; }
+
+    inline std::unique_ptr<TableSchema> copy() const final {
```
Comment (acquamarin) [inline src/include/catalog/table_schema.h:258]: Move copy() below deserialize

**Review by acquamarin — APPROVED** [review]

File: src/binder/bind/bind_ddl.cpp (hunk 2 above)
Comment (andyfengHKU) [reply src/binder/bind/bind_ddl.cpp:79]: Let's keep it here and see if it's needed for querying. If not, we don't need to put in InternalKeyword

File: src/include/catalog/table_schema.h [hunk 9]
```diff
@@ -239,5 +240,35 @@ class RelTableSchema : public TableSchema {
     std::unique_ptr<common::LogicalType> dstPKDataType;
 };
 
+class RdfTableSchema : public TableSchema {
```
[the remark this answers is not in the document]
Comment (andyfengHKU) [reply src/include/catalog/table_schema.h:243]: Done.
P.S. I'm using Rdf instead of RDF everywhere

File: test/test_files/rdf/ddl.test (hunk 5 above)
Comment (andyfengHKU) [reply test/test_files/rdf/ddl.test:6]: I'll leave a TODO here. It's not easy to add manual tax in test file yet.

**Pull request merged**